### PR TITLE
Fixes part of #4195 : Added dark mode support to QuestionPlayer and Exploration

### PR DESCRIPTION
**Pull request by bhaktideshmukh**
## Explanation

Fixes part of #4195 

**Mock link** :-- https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/249c3349-605b-4569-bbe9-1a017ab90101/specs/

This PR handles:--

- Adds dark mode support to input interactions in QuestionPlayer and Exploration
- Handles color of ic_arrow_drop_grey.xml and ic_arrow_right_grey.xml
- Adds dark mode to submitted_answers
- Changed the cursor color

**Screenshots**

| Default | Dark mode |
|-------|---|
|||

| Default | Dark mode |
|-------|---|
|||

| Default | Dark mode |
|-------|---|
|||

| Default | Dark mode |
|-------|---|
|||

| Default | Dark mode |
|-------|---|
|||

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

File: app/src/main/res/drawable/edit_text_background_border_blue.xml [hunk 1]
```diff
@@ -2,8 +2,8 @@
 <shape xmlns:android="http://schemas.android.com/apk/res/android"
   android:shape="rectangle">
   <corners android:radius="4dp" />
-  <solid android:color="@color/color_def_white" />
+  <solid android:color="@color/component_color_shared_text_input_layout_background_color" />
   <stroke
     android:width="2dp"
-    android:color="@color/color_def_oppia_dark_blue" />
+    android:color="@color/component_color_shared_input_interaction_edit_text_border_color" />
```
Comment (rt4914) [inline app/src/main/res/drawable/edit_text_background_border_blue.xml:8]: Why are you naming it like 
`input_interaction_edit_text`
and above you are naming it as `text_input_layout_`

shouldn't these be consistent?

File: app/src/main/res/drawable/submitted_answer_background.xml [hunk 2]
```diff
@@ -4,9 +4,9 @@
   android:shape="rectangle">
   <stroke
     android:width="1dp"
-    android:color="@color/color_def_black"/>
+    android:color="@color/component_color_shared_submitted_answer_item_border_color"/>
   <corners
     android:radius="4dp"/>
   <solid
-    android:color="@color/color_def_white"/>
+    android:color="@color/component_color_shared_submitted_answer_item_background_color"/>
```
Comment (rt4914) [inline app/src/main/res/drawable/submitted_answer_background.xml]: `...._solid_color`

File: app/src/main/res/drawable/submitted_answer_background.xml [hunk 3]
```diff
@@ -4,9 +4,9 @@
   android:shape="rectangle">
   <stroke
     android:width="1dp"
-    android:color="@color/color_def_black"/>
+    android:color="@color/component_color_shared_submitted_answer_item_border_color"/>
```
Comment (rt4914) [inline app/src/main/res/drawable/submitted_answer_background.xml]: `...._stroke_color`

**Review by rt4914 — CHANGES_REQUESTED** [review]
@bhaktideshmukh Please add screenshots for Exploration player too.

File: app/src/main/res/drawable/edit_text_background_border_blue.xml (hunk 1 above)
Comment (bhaktideshmukh) [reply app/src/main/res/drawable/edit_text_background_border_blue.xml:8]: > Why are you naming it like `input_interaction_edit_text` and above you are naming it as `text_input_layout_`
> 
> shouldn't these be consistent?

Correct these should be consistent but in `styles.xml`, for input interaction some are named as `...text_input...` and some as `..input_interaction...` so this was little confusion that which convention should be followed since for already existing color I have not created a new color and have used the existing one.

File: app/src/main/res/drawable/submitted_answer_background.xml (hunk 3 above)
Comment (bhaktideshmukh) [reply app/src/main/res/drawable/submitted_answer_background.xml]: Done

File: app/src/main/res/drawable/submitted_answer_background.xml (hunk 2 above)
Comment (bhaktideshmukh) [reply app/src/main/res/drawable/submitted_answer_background.xml]: Done

**Comment by bhaktideshmukh** [comment]
> @bhaktideshmukh Please add screenshots for Exploration player too.

Done

**Review by rt4914 — CHANGES_REQUESTED** [review]
Make following changes:
1. The hint color should be changed https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/4744bfae-84be-44fa-a27d-9a252fdfc2f9/specs/
2. I don't have any reference for cursor-color but I believe we can use this one https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/d80a87ee-06df-4d82-ab8e-8311b9adccf6/specs/
3. When input-interaction is **not**. selected I think the border should have similar width as that of selected state. Reference: https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/4744bfae-84be-44fa-a27d-9a252fdfc2f9/specs/

**Comment by bhaktideshmukh** [comment]
> Make following changes:
> 
>     1. The hint color should be changed https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/4744bfae-84be-44fa-a27d-9a252fdfc2f9/specs/
> 
>     2. I don't have any reference for cursor-color but I believe we can use this one https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/d80a87ee-06df-4d82-ab8e-8311b9adccf6/specs/
> 
>     3. When input-interaction is **not**. selected I think the border should have similar width as that of selected state. Reference: https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/4744bfae-84be-44fa-a27d-9a252fdfc2f9/specs/

- First is done
- For the cursor color, for now I have changed the color in `theme.xml` since the solution you suggested of using `android:textCursorDrawable` wasn't working. But we can still revert it and can go with some other option.
- Third is done

Have updated the screenshots according to the new changes.

File: app/src/main/res/values-v21/themes.xml [hunk 4]
```diff
@@ -12,7 +12,7 @@
   <style name="OppiaThemeWithoutActionBar" parent="Theme.MaterialComponents.Light.NoActionBar.Bridge">
     <item name="colorPrimary">@color/color_palette_primary_color</item>
     <item name="colorPrimaryDark">@color/color_palette_primary_dark_color</item>
-    <item name="colorAccent">@color/color_palette_accent_color</item>
+    <item name="colorAccent">@color/component_color_shared_edit_text_cursor_color</item>
```
Comment (rt4914) [inline app/src/main/res/values-v21/themes.xml]: This should not be changed here because `colorAccent` must be getting used at various other places too. and it will affect other UI elements when changed.

Changing it in theme is fine but we should be changing the exact `curor_color` to some similar attribute and not its parent attribute like `colorAccent`.

Also the color in lightmode has changed because of latest commit. It should not be black, it should be blue  https://xd.adobe.com/view/3dca36c2-5115-419c-b25e-0f10526b077c-6899/screen/5ef661fa-8995-45b2-b10d-64c1b5cc7890/

**Review by rt4914 — CHANGES_REQUESTED** [review]
@bhaktideshmukh PTAL Thanks.  Somewhere on lines of this solution https://stackoverflow.com/a/7238526

**Comment by bhaktideshmukh** [comment]
@rt4914, I have made the changes the way you suggested but upon clicking input interaction, app is crashing.
I am not getting what's the cause behind this PTAL whats wrong.
Thanks.

**Comment by BenHenning** [comment]
> @rt4914, I have made the changes the way you suggested but upon clicking input interaction, app is crashing. I am not getting what's the cause behind this PTAL whats wrong. Thanks.

Hi @bhaktideshmukh. This sounds like it might be a good case for a debug doc (see https://github.com/oppia/oppia/wiki/Debugging-Docs) as that provides more context as to what you tried, what you're specifically observing, and what you think might be going wrong but are unsure of how to proceed.

Note that it doesn't actually need to be a Google doc--you can add the same information as a comment to this thread to provide context for reviewers. Note that without this information, it can be quite difficult to help since reviewers generally do not know as much about what you've tried as you do.

**Comment by bhaktideshmukh** [comment]
> > @rt4914, I have made the changes the way you suggested but upon clicking input interaction, app is crashing. I am not getting what's the cause behind this PTAL whats wrong. Thanks.
> 
> Hi @bhaktideshmukh. This sounds like it might be a good case for a debug doc (see https://github.com/oppia/oppia/wiki/Debugging-Docs) as that provides more context as to what you tried, what you're specifically observing, and what you think might be going wrong but are unsure of how to proceed.
> 
> Note that it doesn't actually need to be a Google doc--you can add the same information as a comment to this thread to provide context for reviewers. Note that without this information, it can be quite difficult to help since reviewers generally do not know as much about what you've tried as you do.

I guess there was some glitch from my side, I have updated the files and app is not crashing now and is working properly so I don't think there is any need for the debugging doc.

File: app/src/main/res/drawable/color_cursor.xml [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
 <shape xmlns:android="http://schemas.android.com/apk/res/android">
-  <size android:width="0.5dp" />
+  <size android:width="2dp" />
```
Comment (rt4914) [inline app/src/main/res/drawable/color_cursor.xml:3]: Why was this changed? And also how does this affect the light-mode?

File: app/src/main/res/drawable/edit_text_background_border_blue.xml (hunk 1 above)
Comment (rt4914) [reply app/src/main/res/drawable/edit_text_background_border_blue.xml:8]: In that case lets fix this confusion in this PR atleast for this file.

File: app/src/main/res/values-night/color_palette.xml [hunk 6]
```diff
@@ -19,7 +19,7 @@
   <color name="color_palette_highlighted_background_color">@color/color_def_highlight_blue</color>
   <color name="color_palette_light_description_text_color">@color/color_def_white</color>
   <color name="color_palette_primary_background_color">@color/color_def_oppia_light_black</color>
-  <color name="color_palette_hint_text_color">@color/color_def_accessible_light_grey</color>
```
Comment (rt4914) [inline app/src/main/res/values-night/color_palette.xml:22]: Are you sure this was not getting used anywhere else before removing the previous color?

**Review by rt4914 — CHANGES_REQUESTED** [review]
@bhaktideshmukh PTAL at all open comments.

File: app/src/main/res/drawable/color_cursor.xml (hunk 5 above)
Comment (bhaktideshmukh) [reply app/src/main/res/drawable/color_cursor.xml:3]: > Why was this changed? And also how does this affect the light-mode?

Since in develop branch the size for cursor is 2dp which is not handled by this xml file so to keep it consistent I changed the width from 0.5 to 2 dp.

| 0.5dp (current branch) | 2dp (develop branch) |
|-------|---|
|||

File: app/src/main/res/drawable/edit_text_background_border_blue.xml (hunk 1 above)
Comment (bhaktideshmukh) [reply app/src/main/res/drawable/edit_text_background_border_blue.xml:8]: > In that case lets fix this confusion in this PR atleast for this file.

I think `...input_interaction...` sounds better to me. 
 So should I replace `...text_input...` with `...input_interaction...`?

File: app/src/main/res/values-night/color_palette.xml (hunk 6 above)
Comment (bhaktideshmukh) [reply app/src/main/res/values-night/color_palette.xml:22]: > Are you sure this was not getting used anywhere else before removing the previous color?

This color was used in `layout/math_expression_parser_fragment.xml` through `component_colors.xml` but in mocks hint_color was mentioned white thats why I replaced it with white.

**Comment by bhaktideshmukh** [comment]
> @bhaktideshmukh PTAL at all open comments.

Done.

File: app/src/main/res/values-night/color_palette.xml (hunk 6 above)
Comment (rt4914) [reply app/src/main/res/values-night/color_palette.xml:22]: In that case the color for `layout/math_expression_parser_fragment.xml` will get changed and therefore creating an issues. I suggest creating a new color_palette and using that.

Comment (bhaktideshmukh) [reply app/src/main/res/values-night/color_palette.xml:22]: > In that case the color for `layout/math_expression_parser_fragment.xml` will get changed and therefore creating an issues. I suggest creating a new color_palette and using that.

Done

File: app/src/main/res/drawable/color_cursor.xml (hunk 5 above)
Comment (rt4914) [reply app/src/main/res/drawable/color_cursor.xml:3]: Great. Thanks.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.
